13167 docs for connector builder new UI schemaform rollout (#63329)
airbytehq/airbyte: docs/platform/connector-development/connector-builder-ui/authentication.md

- [Basic HTTP](#basic-http)
- [Bearer Token](#bearer-token)
- [API Key](#api-key)

> - [JWT](#jwt)

Reviewer: ```suggestion
- [JWT Authenticator](#jwt)
```

> - [OAuth 2.0 (Declarative)](#oauth-20-declarative)
> - [OAuth (Legacy)](#oauth-legacy)

Reviewer: ```suggestion
- [OAuth2](#oauth-2)
```
The new UI has actually merged the declarative and legacy OAuths into one - now users can utilize Declarative OAuth by selecting the OAuth2 option and then enabling `DeclarativeOAuth Connector Specification` inside it.

  - HS384 (HMAC with SHA-384) 
  - HS512 (HMAC with SHA-512)
  - RS256 (RSA with SHA-256)
  - RS384 (RSA with SHA-384)
  - RS512 (RSA with SHA-512)
  - ES256 (ECDSA with SHA-256)
  - ES384 (ECDSA with SHA-384)
  - ES512 (ECDSA with SHA-512)
- **Base64 Encode Secret Key** - When enabled, the secret key will be base64 encoded before being used to sign the JWT. Only enable this if required by the API.
- **Header Prefix** - The prefix to use in the Authorization header (defaults to "Bearer")
- **JWT Headers** - Additional headers to include in the JWT header object
- **JWT Payload** - Additional properties to include in the JWT payload
- **Additional JWT Headers** - Extra headers to include with the JWT headers object
- **Additional JWT Payload** - Extra properties to add to the JWT payload

#### Example

When fetching records, the JWT token is generated and sent as part of the `Authorization` header:

```
curl -X GET \
  -H "Authorization: Bearer <jwt-token>" \
  https://api.example.com/<stream path>
```

The JWT token is automatically generated by the connector using the configured secret key and algorithm, and includes any additional headers or payload properties you've specified.

> ### OAuth 2.0 (Declarative)

Reviewer: ```suggestion
### OAuth 2
```

> Declarative OAuth 2.0 provides a comprehensive way to configure any OAuth 2.0 flow by describing the exact structure and behavior of the authentication endpoints. This is the recommended approach for implementing OAuth 2.0 authentication in the Connector Builder, as it supports the complete OAuth flow including user consent and automatic token management.

Reviewer: ```suggestion
The OAuth2 authentication method provides a comprehensive way to configure any OAuth 2.0 flow by describing the exact structure and behavior of the authentication endpoints.
```

#### Key Features

- **Complete OAuth 2.0 Flow** - Supports the full authorization code flow with user consent
- **Flexible Configuration** - Accommodates OAuth 2.0 implementations that deviate from standard specifications
- **Automatic Token Management** - Handles access token refresh automatically
- **Custom Parameter Support** - Allows injection of custom parameters and headers at each step

#### Configuration Components

> The Declarative OAuth 2.0 configuration consists of three main components:

Reviewer: This whole section will probably need to be rewritten to reflect what I mentioned above

- **Redirect URI Key** - Custom field name for redirect_uri parameter (defaults to "redirect_uri")
- **Scopes** - Space-separated list of OAuth scopes to request
- **Access Token Key** - Field name in response containing the access token (defaults to "access_token")
- **Access Token Headers** - Custom headers to include in access token requests
- **Access Token Parameters** - Custom query parameters for access token requests

#### Example Configuration

For an API that uses standard OAuth 2.0 with custom parameter names:

- **Consent URL**: `https://api.example.com/oauth/authorize?client_id={{client_id_value}}&redirect_uri={{redirect_uri_value}}&scope={{scope_value}}&state={{state_value}}`
- **Access Token URL**: `https://api.example.com/oauth/token`
- **Scopes**: `read write`

When a user authenticates, they'll be redirected to the consent URL to grant permissions, then the connector will automatically exchange the authorization code for an access token.

To learn more about Declarative OAuth 2.0 and see additional example implementations, please refer to the [Declarative OAuth 2.0](/platform/connector-development/config-based/advanced-topics/oauth) documentation.

> ### OAuth (Legacy)

Reviewer: This section should be combined with the above based on what I mentioned earlier

- **Authenticators**: A mapping of authenticator names to their respective configurations

### How it works

The Selective Authenticator evaluates the value at the specified configuration path and uses it to select the appropriate authenticator from the available options. For example, if the selection path points to a config property with value "api_key", it will use the authenticator mapped to "api_key".

### Example

If you want to support both API Key and OAuth authentication:

1. **Authenticator Selection Path**: `["auth", "type"]`
2. **Authenticators**: 
   - `api_key`: Configure an API Key authenticator
   - `oauth`: Configure an OAuth authenticator

When users configure their connector, they would set `auth.type` to either "api_key" or "oauth", and the Selective Authenticator will automatically use the corresponding authentication method.

### Use Cases

- APIs that support multiple authentication methods
- Connectors that need to work with different API versions requiring different auth
- Providing flexibility for users to choose their preferred authentication method

> 

Reviewer: This isn't really well-supported by the UI today, because we don't support nested inputs in the Inputs tab. Users can only really use this if they switch to YAML mode where they can fully edit the connector spec.

I'd say let's just axe this section for now, or add a note saying that users will need to switch to YAML mode to configure the spec with nested inputs
